marc_to_solr/lib/princeton_marc.rb

```diff
@@ -286,7 +285,7 @@ def process_hierarchy(record, fields, vocabulary = [])
         heading = heading.gsub(" #{s_field.value}", "#{SEPARATOR}#{s_field.value}") if split_on_subfield.include?(s_field.code)
       end
       heading = heading.split(SEPARATOR)
-      heading = change_the_subject.fix(heading)
+      heading = change_the_subject.fix(heading) if ENV['CHANGE_THE_SUBJECT'] == 'true' && record['650'].indicator2 == '0'
```

Comment:
@mzelesky thank you for reviewing and pointing out to limit down to the second indicator 0.